p2p/protocol/head/head_test.go

```diff
@@ -0,0 +1,49 @@
+package head_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-legs/p2p/protocol/head"
+	"github.com/filecoin-project/go-legs/test"
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	_ "github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	"github.com/libp2p/go-libp2p"
+)
+
+func TestFetchLatestHead(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	publisher, _ := libp2p.New(context.Background())
+	client, _ := libp2p.New(context.Background())
+
+	// Provide multiaddrs to connect to
+	client.Peerstore().AddAddrs(publisher.ID(), publisher.Addrs(), time.Hour)
+
+	publisherStore := dssync.MutexWrap(datastore.NewMapDatastore())
+	rootLnk, err := test.Store(publisherStore, basicnode.NewString("hello world"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	p := &head.Publisher{}
+	go p.Serve(ctx, publisher, "test")
```

Comment:
The context passed in here appears to control the lifetime of the server.  Yet the same context is passed to `QueryRootCid`, and that context should be used for cancellation of the function.  However, this test makes it look like the context is used for cancelation of either.

Ultimately, it seems that the context passed into `p.Serve` here does not really do anything since it does not cancel the inner goroutine.